### PR TITLE
Add storage write abstraction and MinIO implementation

### LINKED ISSUE #68: Add storage writer abstraction
### Describe the bug

Add a storage writer abstraction.

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### TrustyAI version or git rev

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by ruivieira**
Ability to save data to abstract storage with a concrete MinIO implementation.

This PR also adds a new method to the `Dataframe` (in order to get just the outputs).
If testing locally, please rebuild the whole of `trustyai-explainability`.

Closes #68

### CODE REVIEW

File: explainability-service/README.md [hunk 1]
```diff
@@ -359,14 +359,30 @@ The data can be batched into the latest `n` observations by using the configurat
...
+
+```shell
+docker build -f ./Dockerfile \
+    -t trustyai/trustyai-service:999-SNAPSHOT .
+```
+
+And test the image with:
+
+```shell
+docker run -i --rm -p 8080:8080 trustyai/trustyai-service:999-SNAPSHOT
```
Comment (danielezonca) [inline explainability-service/README.md]: It should be possible to configure the component to take the Dockerfile to build the image so that we preserve the possibility to build the image with a single mvn command.
Btw let's do it with another PR.
I would suggest for now to remove the `jib` configuration so that we don't risk to have images produced differently

File: explainability-service/src/main/java/org/kie/trustyai/service/data/exceptions/StorageWriteException.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package org.kie.trustyai.service.data.exceptions;
+
+public class StorageWriteException extends Exception {
```
Comment (danielezonca) [inline explainability-service/src/main/java/org/kie/trustyai/service/data/exceptions/StorageWriteException.java]: Let's use unchecked exceptions :)
```suggestion
public class StorageWriteException extends RuntimeException {
```

File: explainability-service/src/main/java/org/kie/trustyai/service/data/storage/MinioStorage.java [hunk 3]
```diff
@@ -95,17 +96,60 @@ public ByteBuffer getInputData() throws StorageReadException {
...
+        final String tempFilename = "tmp-" + UUID.randomUUID();
+        final List<ComposeSource> sources = List.of(
+                ComposeSource.builder().bucket(bucketName).object(filename).build(),
+                ComposeSource.builder().bucket(bucketName).object(tempFilename).build());
+        try {
+            saveData(byteBuffer, bucketName, tempFilename);
+            minioClient.composeObject(ComposeObjectArgs.builder().bucket(bucketName).object(filename)
+                    .sources(sources).build());
+            // Delete temporary file
+            minioClient.removeObject(RemoveObjectArgs.builder().bucket(bucketName).object(tempFilename).build());
```
Comment (danielezonca) [inline explainability-service/src/main/java/org/kie/trustyai/service/data/storage/MinioStorage.java:139]: In theory all of this should be thread safe to avoid race conditions. I would suggest for now to add a `synchronized` keyword to the whole method, knowing that this is going to have impact on performance and that it works only if there is a single replica running

**Comment by ruivieira** [comment]
Build instructions and Dockerfile are being tracked on #72 .

**Review by tteofili — APPROVED** [review]
